sql/analyzer/pushdown.go

```diff
@@ -361,26 +390,20 @@ func pushdownFiltersToTable(
 	filters *filterSet,
 	tableAliases TableAliases,
 ) (sql.Node, error) {
-	if sa, ok := tableNode.(*plan.SubqueryAlias); ok {
```

Comment:
@reltuk I removed this because it wasn't possible for it to be true based on the type case statements at the call sites. Was that a bug? No tests changed when I killed it.

Reply:
Yeah, I don't think this belongs here anymore.

It got added here, in the first pass at pushing this stuff down: https://github.com/dolthub/go-mysql-server/commit/b676d0eb596dc9756e122edda022e470861cc6b0#diff-a0709b6c96e58c2dc069c5d530f1a73d02d0894c2cea77e4ae57d95090e729e0R306

And then it became obsolete here, where the pushdown for subquery aliases got moved to a separate analysis rule: https://github.com/dolthub/go-mysql-server/commit/1582cd4d96562dc0a825c30d26782549681fdd68#diff-a0709b6c96e58c2dc069c5d530f1a73d02d0894c2cea77e4ae57d95090e729e0R196